tests/fx/quantization/test_configuration.py

```diff
@@ -0,0 +1,592 @@
+            )
+        )
+        refs.append(
+            QConfigUnit(
+                dtype=torch.qint8, quant_min=-5, quant_max=5, bins=1024, calibration_method=CalibrationMethod.Histogram
+            )
+        )
+
+        return classes_and_kwargs, refs
+
+    def _test_as_observer_and_fake_quantize(self, as_observer):
+
+        samples = _generate_random_values(100)
+        classes_and_kwargs, refs = self._get_observer_class_and_observer_kwargs()
+
+        outputs = []
+
+        for t, ref in zip(classes_and_kwargs, refs):
+            class_, kwargs = t
+            if as_observer:
+                qconfig_unit = ref.as_observer(as_factory=False)
+                real = class_(**kwargs)
+            else:
+                qconfig_unit = ref.as_fake_quantize(as_factory=False)
+                real = torch.ao.quantization.FakeQuantize(class_, **kwargs)
+
+            for x in samples:
+                qconfig_unit(x)
+                real(x)
+
+            outputs.append((qconfig_unit.calculate_qparams(), real.calculate_qparams()))
+
+        for idx, output in enumerate(outputs):
+            for t1, t2 in zip(*output):
+                self.assertTrue(
+                    torch.allclose(t1, t2),
+                    f"The example do not match between the PyTorch ref and the restored to PyTorch one, max diff: {(t1 - t2).abs().max()}, classes_and_kwargs = {classes_and_kwargs[idx]}, ref = {refs[idx]}",
+                )
+
+    def test_as_observer(self):
+        self._test_as_observer_and_fake_quantize(True)
+
+    def test_as_fake_quantize(self):
+        self._test_as_observer_and_fake_quantize(False)
+
+    def test_from_pytorch_back_to_pytorch(self):
+        samples = _generate_random_values(100)
+        classes_and_kwargs, _ = self._get_observer_class_and_observer_kwargs()
+
+        def test_fn(real_fn, back_fn):
+            outputs = []
+
+            for t in classes_and_kwargs:
+                class_, kwargs = t
+                real = real_fn(class_, kwargs)
+                back = back_fn(real)
+                for x in samples:
+                    real(x)
+                    back(x)
+
+                outputs.append((real.calculate_qparams(), back.calculate_qparams()))
+
+            for idx, output in enumerate(outputs):
+                for t1, t2 in zip(*output):
+                    self.assertTrue(
+                        torch.allclose(t1, t2),
+                        f"The example do not match between the PyTorch ref and the restored to PyTorch one, max diff: {(t1 - t2).abs().max()}, classes_and_kwargs = {classes_and_kwargs[idx]}",
+                    )
+
+        # From observer back to observer
+        test_fn(lambda class_, kwargs: class_(**kwargs), lambda x: QConfigUnit.from_pytorch(x).as_observer(False))
+
+        torch_version = version.parse(torch.__version__)
+        if (torch_version.major, torch_version.minor) >= (1, 12):
```

Comment:
This is not compatible with the constraint on Torch in Transformers no?

Reply:
For now, but could not do otherwise for this specific part.